### ISSUE TITLE
getObject creates reference to underlying object

### ISSUE THREAD

**Issue by rusu24edward** [jpype-project/jpype#407]
## Context
Suppose I have some backend class like this:
``` java
public class TestClass {
    private double myValue;
    public void setValue(double input) { myValue = input; }
    ...
}
```
and I want to wrap this class in order to limit the exposure of TestClass to python, like so:
``` java
public class TestClassWrapper {
    private TestClass internalTestClass;
    TestClassWrapper() {
        internalTestClass = new TestClass(); // Construct empty TestClass
    }
    void setParameter(double input) {
        internalTestClass.setValue(input);
    }
}
```

Now I create the following python script:
``` python
# launch jvm and setup jpype
...
TestClass = jpype.JClass("testpackage.TestClassWrapper")
my_test_class = TestClass()
print dir(my_test_class)
```
I am expecting that my_test_class will have the standard python attributes and the public attributes from TestClassWrapper (and from Object, since it implicitly inherits from Object). It has these, but then it also has an additional attribute called "parameter". This is strange to me since parameter is not defined in TestClassWrapper or in the backend TestClass. Now, when I run these commands:
``` python
print my_test_class.parameter
my_test_class.setParameter(2.0)
print my_test_class.parameter
```
It outputs:
```
0.0
2.0
```
Somehow "parameter" references private TestClass.myValue.

## What I've done
I ran a couple other tests and confirmed that sometimes (not all the time), functions that start with "get" or "set" will result in jpype creating objects that reference the underlying data structure. Is this expected behavior, or is it a bug? I can work around this for now by not using the names "get" and "set", but those are the most natural names to use for the API, so I would like to have that option.

**Comment by Thrameos** [comment]
This is part of the beans modifier class that is currently a default part of the distribution.  It scans for all the get/set methods and creates an automatic property.   The behavior is problematic at times such as the case you are listing.  On the other hand it can be useful if using tab completion as getMethod() does not have a type and thus doesn’t seem to tab complete, but for some reason properties can do so.   Not sure why this is the case, but at least it appears to work that way on python interactive shells such as spyder.

Unfortunately, there is not a lot of documentation on the customizer process.  Whenever a class wrapper is created it consults a hash table to see if there are any customizers for that class.  Most of them are keyed to specific type so that behaviors like “this implements a java.util.Map, thus jpype should implement all the python methods that a map are expected to have.”  The beans one however was completely global and rather random in its function.  Because it was loaded late in the process, many basic java classes such as exceptions were already loaded.  As the customizers are not retroactive, the result is the user sees lots of basic classes without the modifiers, but the user created classes all have it.  It also has interactions with existing fields in the class leading to additional confusion.  Worse still because of a bug in the field creation, jpype also loaded private fields into the namespace.  Meaning it is very hard to tell which was the source of the property.

The current plan for 0.7 is to disable this customizer and relegate it to an optional module.  Thus if you really want this behavior, one would call “import jpype.beans”  otherwise it will do exactly as expected and only expose the public fields and public methods.   I have made this customizer work retroactively so, you don’t have to worry about the order that the class construction.

I am hoping this is an acceptable solution.  There are a few items which unfortunately exposed extra stuff in 0.6 which can at times be useful but at the same time can be problematic or unintuitive.   I wanted to get to a good clean interface as the base and push all that other stuff into legacy support.  For example, JPackage which automatically constructs a package each attempted name lookup is great, but doesn’t understand java naming conventions and thus if a library can’t be found it makes an assumed package rather then finding the class resulting in meaningless errors about package not containing methods.  Thus instead I want to migrate to “from <java.package> import <JavaClass>” which is much safer in that I can report the error appropriately.

**Comment by marscher** [comment]
will be fixed by #418, is that correct @Thrameos?

PS: it would be a good practice to close issues via commits (e.g. write a commit message, which contains 'fixes #number')

**Comment by Thrameos** [comment]
Okay I will try to do that.
